aiomysql/pool.py

```diff
@@ -132,6 +132,17 @@ def acquire(self):
 
     @asyncio.coroutine
     def _fill_free_pool(self, override_min):
+        # iterate over free connections and remove timeouted ones
+        free_size = len(self._free)
+        n = 0
+        while n < free_size:
```

Comment:
What do you think about instead  ``pop/appendleft`` use ``rotate``? This method does not perform unnecessary removals and just relink underlying linked list:

```python
while n < free_size:
    conn = self._free[-1]
    if conn._reader.at_eof():
        self._free.pop()
        conn.close()
    else:
        self._free.rotate()
    n += 1
```